### PR TITLE
Avoid ResourceWarning if shutdown failed for some reason

### PR DESCRIPTION
**Pull request by jolaf**
The following warning appears transiently from time to time while working with websockets:
```python3
/usr/lib/python3/dist-packages/ws4py/websocket.py:230: ResourceWarning: unclosed <ssl.SSLSocket fd=3, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.25.11.39', 40618)>
  self.sock = None
ResourceWarning: Enable tracemalloc to get the object allocation traceback
```
Investigation shows that this warning appears if `self.sock.shutdown(socket.SHUT_RDWR)` (line 225) throws exception and thus `self.sock.close()` (line 226) is not called and execution proceeds straight to `self.sock = None` (line 230), and that produces the `ResourceWarning` mentioned above.

The exception itself is hidden by `pass` statement (line 228), but additional investigation showed the following example of that exception:
```python3
File "/usr/lib/python3/dist-packages/ws4py/websocket.py", line 225, in close_connection
    self.sock.shutdown(socket.SHUT_RDWR)
  File "/usr/lib/python3.8/ssl.py", line 1280, in shutdown
    super().shutdown(how)
OSError: [Errno 107] Transport endpoint is not connected
```
The reliable reproduction of this situation is problematic, but I've created the following test that reproduces the problem reliably and should pass if the problem is fixed:
```python3
from ws4py.client.threadedclient import WebSocketClient
  
def received_message(self, m):
    self.close()

def opened(self):
    self.send('blah\n' * 10000)

def shutdown(self):
    raise Exception('shutdown')

def close_connection(self):
    self.sock.shutdown = shutdown
    original_close_connection(self)

WebSocketClient.received_message = received_message
WebSocketClient.opened = opened

original_close_connection = WebSocketClient.close_connection

WebSocketClient.close_connection = close_connection

ws = WebSocketClient('wss://websocketstest.com/service')
ws.connect()
ws.run_forever()
```
Here's the output:
```shell
$ python3 -W all test.py 
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ws4py/websocket.py", line 225, in close_connection
    self.sock.shutdown(socket.SHUT_RDWR)
  File "test.py", line 10, in shutdown
    raise Exception('shutdown')
Exception: shutdown

/usr/lib/python3/dist-packages/ws4py/websocket.py:231: ResourceWarning: unclosed <ssl.SSLSocket fd=3, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('192.168.1.4', 50110), raddr=('88.198.55.153', 443)>
  self.sock = None
ResourceWarning: Enable tracemalloc to get the object allocation traceback

$ python3
Python 3.8.10 (default, Sep 28 2021, 16:10:42) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ws4py
>>> ws4py.__version__
'0.5.1'

$ uname -a
Linux i-buildcore-01-11 5.4.0-89-generic #100-Ubuntu SMP Fri Sep 24 14:50:10 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

$ cat /etc/issue
Ubuntu 20.04.3 LTS \n \l
```

### CODE REVIEW

**Comment by jolaf** [comment]
I've provided a test to reproduce the problem.

**Review by auvipy — CHANGES_REQUESTED** [review]
is it possible to add the test in the test suite?

**Comment by jolaf** [comment]
> can you just rebase it?

Sorry, I don't get it, what do you mean?

File: ws4py/websocket.py [hunk 1]
```diff
@@ -223,11 +223,13 @@ def close_connection(self):
...
                 self.sock.shutdown(socket.SHUT_RDWR)
+            except:
+                pass
+            try:
                 self.sock.close()
             except:
                 pass
-            finally:
-                self.sock = None
+            self.sock = None
```
Comment (auvipy) [inline ws4py/websocket.py:232]:
```suggestion
            self.sock = None
            
```

**Review by auvipy — CHANGES_REQUESTED** [review]

**Review by auvipy — APPROVED** [review]

**Comment by auvipy** [comment]
no worries, we can include it in next release, can you add/improve test coverage for this change?

**Comment by jolaf** [comment]
Year, next release would be great!

I don't see a good way to improve test coverage in this matter, the test for this is too dirty to be run in some test set.

**Pull request merged**